src/processors.ts

```diff
@@ -86,4 +88,28 @@ export class Processors {
       el.appendChild(pre);
     }
   }
+  
+  public async d3graphvizProcessor(source: string, el: HTMLElement, _: MarkdownPostProcessorContext): Promise<void> {
+    console.debug('Call d3graphvizProcessor');
+    const div = document.createElement('div');
+    const graphId = 'd3graph_' + createHash('md5').update(source).digest('hex').substring(0, 6);
+    div.setAttr('id', graphId);
+    div.setAttr('style', 'text-align: center');
+    el.appendChild(div);
+    const script = document.createElement('script');
+    // graphviz(graphId).renderDot(source); => does not work, ideas how to use it?
+    // Besides, sometimes d3 is undefined, so there must be a proper way to integrate d3.
+    script.text =
+      `if( typeof d3 != 'undefined') { 
+        d3.select("#${graphId}").graphviz()
+        .onerror(d3error)
+       .renderDot(\`${source}\`);
```

Comment:
When I tried to add ` to a label of a node, the dot method works well, but d3 failed without an appropriate error. Could we escape the source before call renderDot so special characters will be handled properly? Thanks.

Reply:
Good point, fixed.